core/domain/question_fetchers.py

```diff
@@ -64,13 +80,16 @@ def get_questions_and_skill_descriptions_by_skill_ids(
     for skill_ids_list in grouped_skill_ids:
         skills = skill_models.SkillModel.get_multi(skill_ids_list)
         grouped_skill_descriptions.append(
-            [skill.description if skill else None for skill in skills])
+            [skill.description if (skill and isinstance(skill.description, str))
+            else None for skill in skills])
```

Comment:
I think this change is not needed because MyPy is able to correct the type at the return type level.

For TODOs, I think TODOs should only be added when we are declaring types explicitly. Something like:
```python
description: str = skill.desciption
```